src/addrman.cpp

```diff
@@ -366,8 +370,11 @@ CAddrInfo CAddrMan::Select_()
             if (GetRandInt(1 << 30) < fChanceFactor * info.GetChance() * (1 << 30))
                 return info;
             fChanceFactor *= 1.2;
+            MilliSleep(100);
         }
     }
+    
```

Comment:
Hi, I kept things as simple as possible as I didn't want to refactor until after a general solution had been accepted.  Also, I don't know if the project intends to rebase against new versions of bitcoin in the future.